### PR TITLE
Fixing spelling issues within the codebase

### LINKED ISSUE #224: Typos and accessibility no-nos
**Describe the bug**
Typos as well as using things like `here` for links make it harder for readers to understand code.

**To Reproduce**
1. [Set up check-spelling](https://github.com/jsoref/sfs-client/commit/2d119c31677af9fe089fdc26a160769fee500b21)
2. View the [results](https://github.com/jsoref/sfs-client/actions/runs/12292227471#summary-34302447496)

**Expected behavior**
No errors relating to typos or forbidden patterns.

**Additional context**
This project is used by winget-cli which uses check-spelling. When I sent an upgrade for it, I was advised that I could send upstream fixes https://github.com/microsoft/winget-cli/pull/5045#discussion_r1878587490.

### PR DESCRIPTION
**Pull request by jsoref**
#### Related Issues

- Closes #224

#### Why is this change being made?

The misspellings have been reported at https://github.com/jsoref/sfs-client/actions/runs/12292227471#summary-34302447496

#### What is being changed?

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling) (which is an evolution of the script I used ages ago when I first made a PR here...).

#### How was the change tested?

The action reports that the changes in this PR would make it happy: 

https://github.com/jsoref/sfs-client/actions/runs/12292227565#summary-34302447544

### CODE REVIEW

File: TEST.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Some test behaviors are controlled by test overrides set via environment variables.
 They only work when the CMake Option `SFS_ENABLE_TEST_OVERRIDES` is set to `ON`.
-To enable that during build, you must use the switch `-EnableTestOverrides` (`--enable-test-overrides` on Linux) with the build scripts. Click [here](README.md#building) for more. See also [here](README.md#building-with-cmake-vscode-extension) for how to set when using VSCode.
+To enable that during build, you must use the switch `-EnableTestOverrides` (`--enable-test-overrides` on Linux) with the build scripts. See [building](README.md#building) for more information. See also [building with cmake vscode extension](README.md#building-with-cmake-vscode-extension) for how to set when using VSCode.
```
Comment (jsoref) [inline TEST.md:5]: This is based on an accessibility complaint (for people using screen readers):

#### Do not use `(click) here` links

For more information, see:
* https://www.w3.org/QA/Tips/noClickHere
* https://webaim.org/techniques/hypertext/link_text
* https://granicus.com/blog/why-click-here-links-are-bad/
* https://heyoka.medium.com/dont-use-click-here-f32f445d1021
```
(?i)(?:>|\[)(?:(?:click |)here|link|(?:read |)more)(?:</|\]\()
```

File: client/tests/functional/details/CurlConnectionTests.cpp [hunk 2]
```diff
@@ -255,7 +255,7 @@ TEST("Testing CurlConnection()")
         REQUIRE_THROWS_CODE(connection->Post(url + "/names", "{}"), HttpNotFound);
 
         url = server.GetBaseUrl() + "/api/v1/contents/" + c_instanceId + "/namespaces/" + c_namespace + "/names/" +
-              c_productName + "/versYions/" + c_version + "/files?action=GenerateDownloadInfo";
+              c_productName + "/versions_/" + c_version + "/files?action=GenerateDownloadInfo";
```
Comment (jsoref) [inline client/tests/functional/details/CurlConnectionTests.cpp:258]: I'm assuming this is functionally equivalent, but it has the advantage of not tripping up spell checking tools.

**Comment by arthuraraujo-msft** [comment]
Hey @jsoref, thanks for the contribution.
Could you please first open a task with more context?

File: TEST.md (hunk 1 above)
Comment (arthuraraujo-msft) [reply TEST.md:5]: Cool!

File: client/tests/functional/details/CurlConnectionTests.cpp (hunk 2 above)
Comment (arthuraraujo-msft) [reply client/tests/functional/details/CurlConnectionTests.cpp:258]: it does, but this was meant to be obvious that something is wrong. The _ is a bit more discrete

File: samples/README.md [hunk 3]
```diff
@@ -4,6 +4,6 @@ This folder contains a few samples that show how to use the SFS Client library.
 
 ## integration-do-client
 
-This sample shows how to use the SFSClient to retrieva metadata and URLs from the SFS Service and the [Delivery Optimization SDK](https://github.com/microsoft/do-client) to perform the download of the retrieved URLs.
+This sample shows how to use the SFSClient to retrieval metadata and URLs from the SFS Service and the [Delivery Optimization SDK](https://github.com/microsoft/do-client) to perform the download of the retrieved URLs.
```
Comment (arthuraraujo-msft) [inline samples/README.md:7]: this was meant to be "retrieve"

File: client/tests/functional/details/CurlConnectionTests.cpp (hunk 2 above)
Comment (jsoref) [reply client/tests/functional/details/CurlConnectionTests.cpp]: Fair. How about moving the `Y` to the end so you have `version` (which is a word) + `Ys` which is short enough that at least check-spelling won't flag it (two letter tokens just aren't worth checking).

**Comment by jsoref** [comment]
I don't think I have time before my next 🛫 to figure out how to make the linter happy. Personally, I hate linters that don't offer a patch output (since they've already done the work, they might as well show their work to save people like me the effort of setting up the tools). If I can figure out how to make the tool do what I want, I might file an issue + PR for that too. But, I don't expect to have much time for that anytime soon. This run of PRs was done while waiting for ✈️s and my vacation is just about over...

File: client/tests/functional/details/CurlConnectionTests.cpp (hunk 2 above)
Comment (arthuraraujo-msft) [reply client/tests/functional/details/CurlConnectionTests.cpp]: Sounds good

**Comment by arthuraraujo-msft** [comment]
> I don't think I have time before my next 🛫 to figure out how to make the linter happy. Personally, I hate linters that don't offer a patch output (since they've already done the work, they might as well show their work to save people like me the effort of setting up the tools). If I can figure out how to make the tool do what I want, I might file an issue + PR for that too. But, I don't expect to have much time for that anytime soon. This run of PRs was done while waiting for ✈️s and my vacation is just about over...

The formatter check is complaining that two files are not properly formatted.

```
The following files have incorrect formatting:

client/include/sfsclient/ClientConfig.h
client/tests/unit/AppContentTests.cpp
```

Do run them through clang-format.
When the project is setup via the setup script, clang-format runs as a pre-commit hook.

**Comment by jsoref** [comment]
The readme has instructions for Windows and Linux. I'm on macOS and the Linux instructions are debian specific and don't detect macOS/not Debian

**Comment by arthuraraujo-msft** [comment]
> The readme has instructions for Windows and Linux. I'm on macOS and the Linux instructions are debian specific and don't detect macOS/not Debian

Sorry, I don't have access to macOS to add instructions for it.
If you have python installed, you could try to run clang format manually.

`pip install clang-format` to install it
`python -c 'import sysconfig; import os; print(sysconfig.get_path("scripts",f"{os.name}_user"))'` to find where it's installed
Then you can call `<path>/clang-format.exe -i <file>` to format the given file

**Comment by arthuraraujo-msft** [comment]
Thanks for the contribution, @jsoref!

**Pull request merged**